sdk/typescript/src/eth-light-client/eth-light-client.ts

```diff
@@ -0,0 +1,305 @@
+ */
+export const createEthereumDWallet = async (
+	dwalletCapId: string,
+	latestEthereumStateId: string,
+	keypair: Keypair,
+	client: DWalletClient,
+) => {
+	const tx = new TransactionBlock();
+	tx.moveCall({
+		target: `${packageId}::${ethDWalletModuleName}::create_eth_dwallet_cap`,
+		arguments: [tx.object(dwalletCapId), tx.object(latestEthereumStateId)],
+	});
+	tx.setGasBudget(200000000);
+
+	let result = await client.signAndExecuteTransactionBlock({
+		signer: keypair,
+		transactionBlock: tx,
+		options: { showEffects: true },
+	});
+
+	if (result.effects?.created?.at(0)?.reference.objectId === undefined) {
+		console.log(result.effects);
+	}
+	return result.effects?.created?.at(0)?.reference.objectId;
+};
+
+/**
+ * Initializes a shared LatestEthereumState object in the dWallet network with the given checkpoint.
+ *
+ * This function should only be called once to initialize the Ethereum state. After the state is initialized,
+ * the Ethereum state object ID is saved, and the state is updated whenever a new state is successfully verified.
+ *
+ * **Logic**
+ * 1. **Select Checkpoint**: Determines the initial checkpoint based on the specified Ethereum network.
+ * 2. **Fetch Bootstrap Data**: Retrieves the bootstrap data required to initialize the Ethereum light client state.
+ * 3. **Initialize State**: Uses the bootstrap data to initialize the Ethereum light client state in BCS format.
+ * 4. **Fetch Updates**: Retrieves updates from the Ethereum consensus RPC since the initial sync period.
+ * 5. **Prepare Transaction**: Constructs a transaction to call the `init_state` function in the Ethereum state module,
+ *    providing the necessary arguments such as the state bytes, network, contract address, and updates.
+ * 6. **Execute Transaction**: Signs and executes the transaction to initialize the Ethereum state on the dWallet network.
+ *
+ * **Arguments**
+ * @param {string} network - The Ethereum network to initialize (e.g., 'mainnet' or 'holesky').
+ * @param {string} rpc - The Ethereum consensus RPC endpoint.
+ * @param {string} contractAddress - The address of the Ethereum smart contract.
+ * @param {number} contractApprovedTxSlot - The slot of the data structure that holds approved transactions in the Ethereum smart contract.
+ * @param {Keypair} keypair - The keypair used to sign the transaction.
+ * @param {DWalletClient} client - The dWallet client instance.
+ */
+export const initEthereumState = async (
+	network: string,
+	rpc: string,
+	contractAddress: string,
+	contractApprovedTxSlot: number,
+	keypair: Keypair,
+	client: DWalletClient,
+) => {
+	let checkpoint = '';
+	switch (network) {
+		case 'mainnet': {
+			checkpoint = '0x8bfa089414dc5fe78dadc8b160a097fe744f17a80251f08eed0a3cdcc60b42f4';
+			break;
+		}
+		case 'holesky': {
+			checkpoint = '0x12e6b81891d23c90502dbc2354de9cb52afe4ff823ca00fd41d411213c6e7bbb';
+			break;
+		}
+		default: {
+			throw new Error('Invalid network');
+		}
+	}
```

Comment:
I think having those checkpoints hard coded is problematic, as after some time those checkpoints will become old, and syncing from them will take a long time.

Reply:
This is correct, but after discussing with Scaly about that, this function should be called once close to genesis of the network, and then all state managing would be handled using the created object.

Yet, another thing I want to add is iterating calls to `verify_new_state` with smaller batches of updates, since at some point the updates array gets too big to pass as a parameter.